src/sqlancer/cnosdb/ast/CnosDBFunction.java

```diff
@@ -0,0 +1,276 @@
+            public boolean supportsReturnType(CnosDBDataType type) {
+                return type == CnosDBDataType.DOUBLE;
+            }
+
+            @Override
+            public CnosDBDataType[] getInputTypesForReturnType(CnosDBDataType returnType, int nrArguments) {
+                return new CnosDBDataType[] { returnType };
+            }
+
+        },
+        LOWER(1, "lower") {
+            @Override
+            public CnosDBConstant apply(CnosDBConstant[] evaluatedArgs, CnosDBExpression... args) {
+                if (evaluatedArgs[0].isNull()) {
+                    return CnosDBConstant.createNullConstant();
+                } else {
+                    String text = evaluatedArgs[0].asString();
+                    return CnosDBConstant.createStringConstant(text.toLowerCase());
+                }
+            }
+
+            @Override
+            public boolean supportsReturnType(CnosDBDataType type) {
+                return type == CnosDBDataType.STRING;
+            }
+
+            @Override
+            public CnosDBDataType[] getInputTypesForReturnType(CnosDBDataType returnType, int nrArguments) {
+                return new CnosDBDataType[] { CnosDBDataType.STRING };
+            }
+
+        },
+        LENGTH(1, "length") {
+            @Override
+            public CnosDBConstant apply(CnosDBConstant[] evaluatedArgs, CnosDBExpression... args) {
+                if (evaluatedArgs[0].isNull()) {
+                    return CnosDBConstant.createNullConstant();
+                }
+                String text = evaluatedArgs[0].asString();
+                return CnosDBConstant.createIntConstant(text.length());
+            }
+
+            @Override
+            public boolean supportsReturnType(CnosDBDataType type) {
+                return type == CnosDBDataType.INT;
+            }
+
+            @Override
+            public CnosDBDataType[] getInputTypesForReturnType(CnosDBDataType returnType, int nrArguments) {
+                return new CnosDBDataType[] { CnosDBDataType.STRING };
+            }
+        },
+        UPPER(1, "upper") {
+            @Override
+            public CnosDBConstant apply(CnosDBConstant[] evaluatedArgs, CnosDBExpression... args) {
+                if (evaluatedArgs[0].isNull()) {
+                    return CnosDBConstant.createNullConstant();
+                } else {
+                    String text = evaluatedArgs[0].asString();
+                    return CnosDBConstant.createStringConstant(text.toUpperCase());
+                }
+            }
+
+            @Override
+            public boolean supportsReturnType(CnosDBDataType type) {
+                return type == CnosDBDataType.STRING;
+            }
+
+            @Override
+            public CnosDBDataType[] getInputTypesForReturnType(CnosDBDataType returnType, int nrArguments) {
+                return new CnosDBDataType[] { CnosDBDataType.STRING };
+            }
+
+        };
+        // NULL_IF(2, "nullif") {
```

Comment:
Perhaps we can remove this commented-out code?

Reply:
Thanks, I remove those comment.